src/main/java/software/amazon/cloudformation/LambdaWrapper.java

```diff
@@ -521,6 +521,29 @@ protected void scrubFiles() {
         return desiredResourceTags;
     }
 
+    /**
+     * Combines the previous tags supplied by the caller (e.g; CloudFormation) into a single
+     * Map which represents the desired final set of tags that were applied to this
+     * resource in the previous state.
+     *
+     * @param request The request object contains the new set of tags to be applied
+     *            at a Stack level. These will be overridden with any resource-level
+     *            tags which are specified as a direct resource property.
+     * @return a Map of Tag names to Tag values
+     */
+    @VisibleForTesting
+    protected Map<String, String> getPreviousResourceTags(final HandlerRequest<ResourceT, CallbackT> request) {
+        Map<String, String> previousResourceTags = new HashMap<>();
+
+        if (request != null && request.getRequestData() != null) {
+            replaceInMap(previousResourceTags, request.getRequestData().getPreviousStackTags());
+            replaceInMap(previousResourceTags,
```

Comment:
The `getPreviousResourceProperties()` is null for CREATE/DELETE/READ,  and not all `provideResourceDefinedTags` handles the null resource model. let's have a null check here.

```
if (request.getRequestData().getPreviousResourceProperties() != null) {
 replaceInMap(previousResourceTags,provideResourceDefinedTags(request.getRequestData().getPreviousResourceProperties()));
}

```